modules/artifact/src/main/java/org/mule/module/artifact/classloader/IBMMQResourceReleaser.java

```diff
@@ -388,4 +398,50 @@ private void processThreadLocalMap(Field threadLocalMapTableField, Object thread
       }
     }
   }
+
+  /**
+   * Close JMSCCThreadPoolMaster threads pending to close to avoid thread leaks after redeployment
+   */
+  private void disposeMQThreads() {
+    List<Thread> threads = ThreadUtils.getAllThreads().stream()
+        .filter(thread -> thread.getName().equals(JMSCC_THREAD_POOL_MAIN_NAME))
+        .collect(Collectors.toList());
+
+    if (!threads.isEmpty()) {
+      threads.forEach(thread -> killThread(thread));
+      try {
+        killIBMWorker();
+      } catch (Throwable e) {
+        LOGGER.debug("An error occurred trying to close the WorkQueueManager", e);
+      }
+    }
+  }
+
+  /**
+   * Kill any thread alive
+   *
+   * @param thread thread to kill
+   */
+  private void killThread(Thread thread) {
+    try {
+      Class<? extends Thread> threadClass = thread.getClass();
+      Method closeMethod = threadClass.getDeclaredMethod("close");
+      closeMethod.setAccessible(true);
+      closeMethod.invoke(thread);
+      LOGGER.info("Thread name : " + thread.getName());
+      thread.interrupt();
+    } catch (Throwable e) {
+      LOGGER.debug("An error occurred trying to close the '" + JMSCC_THREAD_POOL_MAIN_NAME + "' Thread", e);
+    }
+  }
```

Comment:
I'm not entirely sure that calling `close` on the `WorkQueueManagerThread` is necessary because it seems the `WorkQueueManager#close` already does that. Unless you tell me that in a previous version of the driver it wasn't like that. If that's the case we can probably encourage the customer to use a more recent version.

Avoiding this would be very important for Java 17 support because that inner class class is private and it wouldn't work without opening the module.

Finally, letting these threads finish doesn't really remove the static reference to them via `WorkQueueManager` -> `WorkQueueManagerImplementation manager` -> `WorkQueueManagerThread workManagerThread`. So there will still be a leak of the classloader of the artifact the connection resides in, unless it is the same as the one that loaded the `WorkQueueManager` class (which is usually the CL that has the driver as dep).